scripts/zap-scanner.sh

```diff
@@ -37,60 +34,74 @@ elif [ "$ENVIRONMENT" = "local" ]; then
 else
     echo "Invalid environment $ENVIRONMENT"
     exit 1
+fi
+
+cd "$TARGET_DIR" || exit 2
 
+# Ensure the APP_URL is reachable from the zaproxy container
+if ! docker-compose run --rm zaproxy curl -Is "$APP_URL" > /dev/null 2>&1; then
+  echo "Target application at $APP_URL is unreachable by ZAP scanner"
+  exit 3
 fi
 
+echo "================== OWASP ZAP tests =================="
+
+# Ensure the reports directory can be written to
+chmod 777 "$(pwd)"/reports
 
-# do an OWASP ZAP scan
-export ZAP_CONFIG=" \
+# Command line options to the ZAP application
+ZAP_CLI_OPTIONS="\
   -config globalexcludeurl.url_list.url\(0\).regex='.*/robots\.txt.*' \
   -config globalexcludeurl.url_list.url\(0\).description='Exclude robots.txt' \
   -config globalexcludeurl.url_list.url\(0\).enabled=true \
   -config spider.postform=true"
 
-echo "================== OWASP ZAP tests =================="
-cd $TARGET_DIR
-
-if [ "$TARGET" = "frontend" ]; then
-    docker-compose down
-    docker-compose up -d --build
-fi
-
-# Ensure the reports directory can be written to
-chmod 777 $(pwd)/reports
+# How long ZAP will crawl the app with the spider process
+ZAP_SPIDER_MINS=5
 
+ZAP_ARGS=(-t "$APP_URL" -m "$ZAP_SPIDER_MINS" -r "$REPORT_NAME" -z "$ZAP_CLI_OPTIONS")
```

Comment:
If we build the arguments using an array, we can dynamically pass them in to the docker compose command without needing to define the whole command twice.